../../../../connectors/source-google-workspace-admin-reports/acceptance-test-config.yml
/Users/ella/airbytehq/airbyte/airbyte-integrations/connectors/source-google-workspace-admin-reports
============================= test session starts ==============================
platform linux -- Python 3.9.11, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
rootdir: /test_input
plugins: timeout-1.4.2, hypothesis-6.54.6, requests-mock-1.9.3, cov-3.0.0, sugar-0.9.6, mock-3.6.1
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
collected 36 items / 34 deselected / 1 skipped / 1 selected

../../test_input/test_core.py E                                          [ 50%]
../../test_input/test_incremental.py E                                   [100%]


==================================== ERRORS ====================================
______________ ERROR at setup of TestBasicRead.test_read[inputs0] ______________

connector_config = SecretDict(******)
docker_runner = <connector_acceptance_test.utils.connector_runner.ConnectorRunner object at 0xffffa6f87d30>
cached_schemas = {}, cache_discovered_catalog = True

    @pytest.fixture(name="discovered_catalog")
    def discovered_catalog_fixture(
        connector_config, docker_runner: ConnectorRunner, cached_schemas, cache_discovered_catalog: bool
    ) -> MutableMapping[str, AirbyteStream]:
        """JSON schemas for each stream"""
        cached_schemas = cached_schemas.setdefault(make_hashable(connector_config), {})
        if not cache_discovered_catalog:
            cached_schemas.clear()
        if not cached_schemas:
>           output = docker_runner.call_discover(config=connector_config)

connector_acceptance_test/conftest.py:290: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
connector_acceptance_test/utils/connector_runner.py:91: in call_discover
    output = list(self.run(cmd=cmd, config=config, **kwargs))
connector_acceptance_test/utils/connector_runner.py:120: in run
    for line in self.read(container, command=cmd, with_ext=raise_container_error):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'connector_acceptance_test.utils.connector_runner.ConnectorRunner'>
container = <Container: 1b12f459ca>
command = 'discover --config /data/tap_config.json', with_ext = True

    @classmethod
    def read(cls, container: Container, command: str = None, with_ext: bool = True) -> Iterable[str]:
        """Reads connector's logs per line"""
        buffer = b""
        exception = ""
        line = ""
        for chunk in container.logs(stdout=True, stderr=True, stream=True, follow=True):
    
            buffer += chunk
            while True:
                # every chunk can include several lines
                found = buffer.find(b"\n")
                if found <= -1:
                    break
    
                line = buffer[: found + 1].decode("utf-8")
                if len(exception) > 0 or line.startswith("Traceback (most recent call last)"):
                    exception += line
                else:
                    yield line
                buffer = buffer[found + 1 :]
    
        if buffer:
            # send the latest chunk if exists
            line = buffer.decode("utf-8")
            if exception:
                exception += line
            else:
                yield line
        try:
            exit_status = container.wait()
            container.remove()
        except NotFound as err:
            logging.error(f"Waiting error: {err}, logs: {exception or line}")
            raise
        if exit_status["StatusCode"]:
            error = exit_status.get("Error") or exception or line
            logging.error(f"Docker container failed, " f'code {exit_status["StatusCode"]}, error:\n{error}')
            if with_ext:
>               raise ContainerError(
                    container=container,
                    exit_status=exit_status["StatusCode"],
                    command=command,
                    image=container.image,
                    stderr=error,
                )
E               docker.errors.ContainerError: Command 'discover --config /data/tap_config.json' in image '<Image: 'airbyte/source-google-workspace-admin-reports:dev'>' returned non-zero exit status 1: Traceback (most recent call last):
E                 File "/airbyte/integration_code/main.py", line 9, in <module>
E                   from source_google_workspace_admin_reports import SourceGoogleWorkspaceAdminReports
E                 File "/airbyte/integration_code/source_google_workspace_admin_reports/__init__.py", line 1, in <module>
E                   from .source import SourceGoogleWorkspaceAdminReports
E                 File "/airbyte/integration_code/source_google_workspace_admin_reports/source.py", line 8, in <module>
E                   from .client import Client
E                 File "/airbyte/integration_code/source_google_workspace_admin_reports/client.py", line 10, in <module>
E                   from .api import API, AdminAPI, DriveAPI, IncrementalStreamAPI, LoginsAPI, MeetAPI, MobileAPI, OAuthTokensAPI
E                 File "/airbyte/integration_code/source_google_workspace_admin_reports/api.py", line 15, in <module>
E                   import pytz
E               ModuleNotFoundError: No module named 'pytz'

connector_acceptance_test/utils/connector_runner.py:174: ContainerError
---------------------------- Captured stdout setup -----------------------------
{"type": "LOG", "log": {"level": "WARN", "message": "The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format."}}
{"type": "LOG", "log": {"level": "ERROR", "message": "Docker container failed, code 1, error:\nTraceback (most recent call last):\n  File \"/airbyte/integration_code/main.py\", line 9, in <module>\n    from source_google_workspace_admin_reports import SourceGoogleWorkspaceAdminReports\n  File \"/airbyte/integration_code/source_google_workspace_admin_reports/__init__.py\", line 1, in <module>\n    from .source import SourceGoogleWorkspaceAdminReports\n  File \"/airbyte/integration_code/source_google_workspace_admin_reports/source.py\", line 8, in <module>\n    from .client import Client\n  File \"/airbyte/integration_code/source_google_workspace_admin_reports/client.py\", line 10, in <module>\n    from .api import API, AdminAPI, DriveAPI, IncrementalStreamAPI, LoginsAPI, MeetAPI, MobileAPI, OAuthTokensAPI\n  File \"/airbyte/integration_code/source_google_workspace_admin_reports/api.py\", line 15, in <module>\n    import pytz\nModuleNotFoundError: No module named 'pytz'\n"}}
------------------------------ Captured log setup ------------------------------
WARNING  root:config.py:282 The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format.
ERROR    root:connector_runner.py:172 Docker container failed, code 1, error:
Traceback (most recent call last):
  File "/airbyte/integration_code/main.py", line 9, in <module>
    from source_google_workspace_admin_reports import SourceGoogleWorkspaceAdminReports
  File "/airbyte/integration_code/source_google_workspace_admin_reports/__init__.py", line 1, in <module>
    from .source import SourceGoogleWorkspaceAdminReports
  File "/airbyte/integration_code/source_google_workspace_admin_reports/source.py", line 8, in <module>
    from .client import Client
  File "/airbyte/integration_code/source_google_workspace_admin_reports/client.py", line 10, in <module>
    from .api import API, AdminAPI, DriveAPI, IncrementalStreamAPI, LoginsAPI, MeetAPI, MobileAPI, OAuthTokensAPI
  File "/airbyte/integration_code/source_google_workspace_admin_reports/api.py", line 15, in <module>
    import pytz
ModuleNotFoundError: No module named 'pytz'
____ ERROR at setup of TestIncremental.test_read_sequential_slices[inputs0] ____

connector_config = SecretDict(******)
docker_runner = <connector_acceptance_test.utils.connector_runner.ConnectorRunner object at 0xffffa7ae7e20>
cached_schemas = {}, cache_discovered_catalog = True

    @pytest.fixture(name="discovered_catalog")
    def discovered_catalog_fixture(
        connector_config, docker_runner: ConnectorRunner, cached_schemas, cache_discovered_catalog: bool
    ) -> MutableMapping[str, AirbyteStream]:
        """JSON schemas for each stream"""
        cached_schemas = cached_schemas.setdefault(make_hashable(connector_config), {})
        if not cache_discovered_catalog:
            cached_schemas.clear()
        if not cached_schemas:
>           output = docker_runner.call_discover(config=connector_config)

connector_acceptance_test/conftest.py:290: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
connector_acceptance_test/utils/connector_runner.py:91: in call_discover
    output = list(self.run(cmd=cmd, config=config, **kwargs))
connector_acceptance_test/utils/connector_runner.py:120: in run
    for line in self.read(container, command=cmd, with_ext=raise_container_error):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'connector_acceptance_test.utils.connector_runner.ConnectorRunner'>
container = <Container: 3b78686ec2>
command = 'discover --config /data/tap_config.json', with_ext = True

    @classmethod
    def read(cls, container: Container, command: str = None, with_ext: bool = True) -> Iterable[str]:
        """Reads connector's logs per line"""
        buffer = b""
        exception = ""
        line = ""
        for chunk in container.logs(stdout=True, stderr=True, stream=True, follow=True):
    
            buffer += chunk
            while True:
                # every chunk can include several lines
                found = buffer.find(b"\n")
                if found <= -1:
                    break
    
                line = buffer[: found + 1].decode("utf-8")
                if len(exception) > 0 or line.startswith("Traceback (most recent call last)"):
                    exception += line
                else:
                    yield line
                buffer = buffer[found + 1 :]
    
        if buffer:
            # send the latest chunk if exists
            line = buffer.decode("utf-8")
            if exception:
                exception += line
            else:
                yield line
        try:
            exit_status = container.wait()
            container.remove()
        except NotFound as err:
            logging.error(f"Waiting error: {err}, logs: {exception or line}")
            raise
        if exit_status["StatusCode"]:
            error = exit_status.get("Error") or exception or line
            logging.error(f"Docker container failed, " f'code {exit_status["StatusCode"]}, error:\n{error}')
            if with_ext:
>               raise ContainerError(
                    container=container,
                    exit_status=exit_status["StatusCode"],
                    command=command,
                    image=container.image,
                    stderr=error,
                )
E               docker.errors.ContainerError: Command 'discover --config /data/tap_config.json' in image '<Image: 'airbyte/source-google-workspace-admin-reports:dev'>' returned non-zero exit status 1: Traceback (most recent call last):
E                 File "/airbyte/integration_code/main.py", line 9, in <module>
E                   from source_google_workspace_admin_reports import SourceGoogleWorkspaceAdminReports
E                 File "/airbyte/integration_code/source_google_workspace_admin_reports/__init__.py", line 1, in <module>
E                   from .source import SourceGoogleWorkspaceAdminReports
E                 File "/airbyte/integration_code/source_google_workspace_admin_reports/source.py", line 8, in <module>
E                   from .client import Client
E                 File "/airbyte/integration_code/source_google_workspace_admin_reports/client.py", line 10, in <module>
E                   from .api import API, AdminAPI, DriveAPI, IncrementalStreamAPI, LoginsAPI, MeetAPI, MobileAPI, OAuthTokensAPI
E                 File "/airbyte/integration_code/source_google_workspace_admin_reports/api.py", line 15, in <module>
E                   import pytz
E               ModuleNotFoundError: No module named 'pytz'

connector_acceptance_test/utils/connector_runner.py:174: ContainerError
---------------------------- Captured stdout setup -----------------------------
{"type": "LOG", "log": {"level": "ERROR", "message": "Docker container failed, code 1, error:\nTraceback (most recent call last):\n  File \"/airbyte/integration_code/main.py\", line 9, in <module>\n    from source_google_workspace_admin_reports import SourceGoogleWorkspaceAdminReports\n  File \"/airbyte/integration_code/source_google_workspace_admin_reports/__init__.py\", line 1, in <module>\n    from .source import SourceGoogleWorkspaceAdminReports\n  File \"/airbyte/integration_code/source_google_workspace_admin_reports/source.py\", line 8, in <module>\n    from .client import Client\n  File \"/airbyte/integration_code/source_google_workspace_admin_reports/client.py\", line 10, in <module>\n    from .api import API, AdminAPI, DriveAPI, IncrementalStreamAPI, LoginsAPI, MeetAPI, MobileAPI, OAuthTokensAPI\n  File \"/airbyte/integration_code/source_google_workspace_admin_reports/api.py\", line 15, in <module>\n    import pytz\nModuleNotFoundError: No module named 'pytz'\n"}}
------------------------------ Captured log setup ------------------------------
ERROR    root:connector_runner.py:172 Docker container failed, code 1, error:
Traceback (most recent call last):
  File "/airbyte/integration_code/main.py", line 9, in <module>
    from source_google_workspace_admin_reports import SourceGoogleWorkspaceAdminReports
  File "/airbyte/integration_code/source_google_workspace_admin_reports/__init__.py", line 1, in <module>
    from .source import SourceGoogleWorkspaceAdminReports
  File "/airbyte/integration_code/source_google_workspace_admin_reports/source.py", line 8, in <module>
    from .client import Client
  File "/airbyte/integration_code/source_google_workspace_admin_reports/client.py", line 10, in <module>
    from .api import API, AdminAPI, DriveAPI, IncrementalStreamAPI, LoginsAPI, MeetAPI, MobileAPI, OAuthTokensAPI
  File "/airbyte/integration_code/source_google_workspace_admin_reports/api.py", line 15, in <module>
    import pytz
ModuleNotFoundError: No module named 'pytz'
=============================== warnings summary ===============================
connector_acceptance_test/config.py:282: 36 warnings
test_core.py: 1 warning
  /airbyte/connector_acceptance_test/connector_acceptance_test/config.py:282: DeprecationWarning: The 'warn' function is deprecated, use 'warning' instead
    logging.warn("The acceptance-test-config.yml file is in a legacy format. Please migrate to the latest format.")

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
ERROR ../../test_input/test_core.py::TestBasicRead::test_read[inputs0] - dock...
ERROR ../../test_input/test_incremental.py::TestIncremental::test_read_sequential_slices[inputs0]
SKIPPED [1] connector_acceptance_test/plugin.py:63: Skipping TestFullRefresh.test_sequential_reads: not found in the config.
=========== 1 skipped, 34 deselected, 37 warnings, 2 errors in 3.60s ===========
